packetbeat/protos/smtp/smtp_test.go

```diff
@@ -0,0 +1,205 @@
+// +build !integration
+
+package smtp
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func TestSmtpParser_startResponse(t *testing.T) {
```

Comment:
please use camel case for test names only. Without underscore. I know other modules may still use `_` in names, but this is some minor technical debt not being cleaned up yet.

e.g `TestSmtpParser_startResponse -> TestParserStartResponse` 

If possible consider more tabular tests for testing some common functionality (only use tabular testing if it's more convenient and reduces common boilerplate). E.g.:

```
func TestParser(t *testing.T) {
  var tests := []struct{
    name string
    pub *transPub
    inputs []string
    validate func(*testing.T, pub *transPub, *message)
  } {
  }
 
  for _, test := range tests {
    test := test // so name is unique within scope
    t.Run(test.name, func(t *testing.T) {
      ...
    })
  }
}
```

The `validate` is a function + you can compose multiple function (using closures) to build more enhance validators. If you want to validate per input consider replacing `inputs` and `validate` with a struct like

```
type testInput struct {
  content string
  validate func(...)
}
```